nifi-nar-bundles/nifi-graph-bundle/nifi-graph-processors/src/main/java/org/apache/nifi/processors/graph/ExecuteGraphQueryRecord.java

```diff
@@ -0,0 +1,265 @@
+        ) {
+            Record record;
+
+            long start = System.currentTimeMillis();
+            long recordIndex = 0;
+            long batchStart = start;
+            String recordScript = getScript(SUBMISSION_SCRIPT, context, input);
+            writer.beginRecordSet();
+            while ((record = reader.nextRecord()) != null) {
+                ConcurrentMap<String, List<Object>> dynamicPropertyMap = new ConcurrentHashMap<>();
+                for (PropertyDescriptor entry : context.getProperties().keySet()) {
+                    if(entry.isDynamic()) {
+                        if(!dynamicPropertyMap.containsKey(entry.getName())) {
+                            String valueRecordPath = context.getProperty(entry.getName()).evaluateAttributeExpressions(input).getValue();
+                            final RecordPath recordPath = recordPathCache.getCompiled(valueRecordPath);
+                            final RecordPathResult result = recordPath.evaluate(record);
+                            List<Object> resultSet = result.getSelectedFields().filter(fv -> fv.getValue() != null).map(FieldValue::getValue).collect( Collectors.toList());
+                            dynamicPropertyMap.put(entry.getName(), resultSet);
+                        }
+                    }
+                }
+                Map<String, Object> parameters = new HashMap<>(dynamicPropertyMap);
+
+                parameters.putAll(input.getAttributes());
+
+                graphResponses.addAll(executeQuery(recordScript, parameters));
+
+                recordIndex++;
+                if (recordIndex % 100 == 0) {
+                    long now = System.currentTimeMillis();
+                    batchStart = System.currentTimeMillis();
+                }
+            }
+            writer.finishRecordSet();
+            long end = System.currentTimeMillis();
+            delta = (end - start) / 1000;
+            if (getLogger().isDebugEnabled()){
+                getLogger().debug(String.format("Took %s seconds.", delta));
+            }
+        } catch (Exception ex) {
+            getLogger().error("", ex);
+            failed = true;
+        } finally {
+            if (failed) {
+                session.remove(output);
+                session.getProvenanceReporter().route(input, REL_FAILURE);
+                session.transfer(input, REL_FAILURE);
+            } else {
+                input = session.putAttribute(input, GRAPH_OPERATION_TIME, String.valueOf(delta));
+                session.getProvenanceReporter().send(input, clientService.getTransitUrl(), delta*1000);
+                session.transfer(input, REL_SUCCESS);
+                FlowFile graph = session.create(input);
+                session.getProvenanceReporter().clone(input, graph);
+                boolean error = false;
+                try (OutputStream os = session.write(graph)) {
```

Comment:
My gut tells me that you can refactor this block so that a lot of it happens in the main try and the catch/finally here as well.

Reply:
You are correct. Refactored and put it into the main try-catch block.